### PR TITLE
Use soft-assertions in jaxrs-tests

### PR DESCRIPTION
**Pull request by snazy**
Soft-assertions were really helpful while working on the storage-stuff, so I think, it makes sense to add them to the main code base.

### CODE REVIEW

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRest.java [hunk 1]
```diff
@@ -60,6 +65,10 @@ public NessieApiV1 getApi() {
 
   @AfterEach
   public void tearDown() throws Exception {
+    // Cannot use @ExtendWith(SoftAssertionsExtension.class) + @InjectSoftAssertions here, because
+    // of Quarkus class loading issues. See https://github.com/quarkusio/quarkus/issues/19814
+    soft.assertAll();
```
Comment (dimas-b) [inline servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRest.java:70]: Will this prevent test branch cleanup (below) if it throws? Could we put it in a separate `@AfterEach` method instead?

**Review by dimas-b — COMMENTED** [review]
LGTM overall. Just one concern with test data cleanup.

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRest.java (hunk 1 above)
Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRest.java:70]: Ah, good catch

Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRest.java]: try-finally works, too

**Review by dimas-b — APPROVED** [review]

**Pull request merged**